main.py

```diff
@@ -36,21 +64,32 @@ def api_text_to_speech():
     f.write(data)
     f.close()
 
-    result = model.transcribe(filename)
+    max_attempts = 5
+    for attempt in range(max_attempts):
+        result = model.transcribe(filename, initial_prompt=' '.join(hint_phrases))
+        processed_text = clean_text(result['text'])
+        if processed_text:
+            break
```

Comment:
It seemed like a good idea, but now that you mention it, I don't see why Whisper wouldn't be deterministic. I will test that.

Reply:
I found [this](https://github.com/openai/whisper/discussions/81) thread, and apparently Whisper is, in fact, not deterministic. Again, I will test it myself. If the results worsen with each interaction, The loop should be removed entirely.

Reply:
With hint_phrases given does Whisper only restrict itself to returning these hint phrases ? Here you are checking if the processed_text is defined to break the loop. I think in most cases it will be even if it does not match the hint phrases. 

I'm okay with having a loop if it is not deterministic but if we put one :
- The number of iterations should be configurable
- The condition to break out of the loop should be that we find an exact match from the hint phrases.